pages/project/[id]/index.tsx

```diff
@@ -131,6 +131,55 @@ function ProjectPage({ projectId }: { projectId: string }) {
   const projectQuery = trpc.project.detail.useQuery(projectQueryInput)
   const project = projectQuery.data
 
+  // calculate projectBelongsToUser at the component level
+  let projectBelongsToUser = false
+  // if we have information about the project and if the user is logged in,
+  // compare the IDs of the project creator and the current user
+  // if these are the same, projectBelongsToUser becomes "true"
+  if (project && session) {
+    projectBelongsToUser = project.author.id === session.user.id
+  }
+
+  const [activeTab, setActiveTab] = React.useState<string | null>(null)
+  const [activeCommentsTab, setActiveCommentsTab] = React.useState<
+    string | null
+  >(null)
+
+  // QUESTION: should value be set to a more specific type?
```

Comment:
Perhaps, but it also works like this. :blush: I imagine the next step could be to move this logic into a wrapper around the taps component so that it becomes available on all pages that use tabs, but that can wait until we have more usage on the platform and people use the feature. I'll find it very convenient to be able to just link people to their preferences tab when they ask where they can adjust their preferences! :smiley:

Reply:
ooh ok yeah! maybe this would be good for me to do before setting this up for the project page filters

Reply:
Potentially! But let's hold off on it until we have more of a direction again. Matt wants to do some interviews to see what else we can help with, so I'm not currently confident enough that it'll be worth your while.